akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentImpl.scala

```diff
@@ -140,4 +144,17 @@ private[impl] final class AgentImpl[A <: Agent](
 
   }
 
+  override def transformResponse(modelResponse: String, responseType: Class[_]): BytesPayload = {
+    if (responseType == classOf[String]) {
+      new BytesPayload(ByteString.fromString(modelResponse), serializer.contentTypeFor(responseType))
```

Comment:
This doesn't work yet. Fails when deserializing in the client, since it's not really json, but plain text.